lib/api/CMakeLists.txt

```diff
@@ -139,6 +139,8 @@ foreach(variant IN ITEMS client server)
     "${CMAKE_CURRENT_SOURCE_DIR}/src/rc_check_auth_credentials.cpp"
     "${CMAKE_CURRENT_SOURCE_DIR}/src/rc_data_object_finalize.cpp"
     "${CMAKE_CURRENT_SOURCE_DIR}/src/rc_data_object_modify_info.cpp"
+    "${CMAKE_CURRENT_SOURCE_DIR}/src/rc_delay_rule_tag.cpp"
```

Comment:
consider 'delay_rule_claim' rather than 'delay_rule_tag'?   i know there are 106 places in this one commit... but note sure 'tag' is the right word.   definitely can discuss.

Reply:
'delay_rule_remove_claim' is probably most grammatically correct.

i don't hate 'tag'... it's generic...  but right now, it's a single use tagging system... just 'claiming' it for a particular host.

Reply:
that said - is there anything else we can think through at the moment to 'tag' a rule?    no other use cases...

Reply:
also could just be 'update_status'.     ?

Reply:
> that said - is there anything else we can think through at the moment to 'tag' a rule? no other use cases...

Nothing comes to mind. Would need to think on that one.

> also could just be 'update_status'. ?

Very true.

I've been wondering if we should add a new column instead of repurposing the status column. Doing that means GenQuery1/2 would need to grow support for it, but the benefit is that admins and new users aren't confused by the mismatch of the status column name and its value.

Reply:
> I've been wondering if we should add a new column instead of repurposing the status column.

i don't think we've ever 'used' that status field before?    hmm....

Reply:
That's likely true, but there's nothing saying we may want to eventually use the status column for true status information.

Reply:
yep.  true.    this *is* a status... so it's not wrong... but i understand your point.

what would you name a new field?

Reply:
The presence of the hostname means **a** delay server instance has started **executing** the delay rule. For that reason, how do you feel about the following:
- execution_host
- exe_host
- executor_host
- launched_by
- executed_by
- execution_manager
- execution_owner
- managed_by

Are there other names?

Keep in mind that value of the column can change depending on the delay hints.

Reply:
good list...

'execution_host' is strong.

also...
'claimed_by'

are we also putting a timestamp in there?   so host and time... and ... delay server pid?   hmmm

Reply:
i also like `executor_host`... since that's what we call them in the configuration listing.

Reply:
`claimed_by` is a good one too. I like the use of `execut*` in the word for the same reasons. These are good options. Maybe example usage will help us pick one.
```bash
iquery "select DELAY_RULE_ID, DELAY_RULE_EXECUTION_HOST"
iquery "select DELAY_RULE_ID, DELAY_RULE_EXECUTOR_HOST"
iquery "select DELAY_RULE_ID, DELAY_RULE_CLAIMED_BY"
```

> are we also putting a timestamp in there? so host and time... and ... delay server pid?

No, but I think all of that is important to track for forensics if nothing else.

Reply:
leaning back to `DELAY_RULE_EXECUTION_HOST`

Reply:
I just realized that the use of `executor` for the db column name isn't correct because that represents the server running the agent responsible for doing the work.

Decided to go with the following db column names:
- delay_server_host
- delay_server_pid
- claimed_ts

Reply:
A delay rule is only claimed if the 3 new db columns hold empty strings.
- delay_server_host
- delay_server_pid
- claimed_ts

Reply:
that seems to work.    go go go.

Reply:
Here are the new GenQuery column names.

#### GenQuery1
- `RULE_EXEC_DELAY_SERVER_HOST`
- `RULE_EXEC_DELAY_SERVER_PID`
- `RULE_EXEC_DELAY_SERVER_CLAIMED_TIME`

#### GenQuery2
- `DELAY_RULE_DELAY_SERVER_HOST`
- `DELAY_RULE_DELAY_SERVER_PID`
- `DELAY_RULE_DELAY_SERVER_CLAIMED_TIME`

Reply:
It could, but then it isn't obvious that it's related to the other columns.

I've wondered whether all the column names should be changed to include `claim`. For example:
- `DELAY_RULE_CLAIM_HOST`
- `DELAY_RULE_CLAIM_HOST_PID`
- `DELAY_RULE_CLAIM_TIME`

Reply:
i don't hate those three...

and so...

GenQuery1
 - RULE_EXEC_CLAIM_HOST
 - RULE_EXEC_CLAIM_HOST_PID
 - RULE_EXEC_CLAIM_TIME

Reply:
Ok. Will update the PR to use these names.

Reply:
Also updated db column names to match the GenQuery column names.
- `claim_host`
- `claim_host_pid`
- `claim_ts`

Done.